### ISSUE TITLE
Sensu for Mac crashes with ruby illegal instructions 4 on some Intel CPU...

### ISSUE THREAD

**Issue by patrickatomic** [sensu/sensu#1623]
Installing sensu on many mac machines worked fine! But looks like the pre-compiled version of Ruby on some Mac doesn't support the CPU they have... Same Mac version with different CPU won;'t have the same output, and 

here's the output while doing a check (but no matter which check I'm doing, I have the same result on the machines that are failing...)
```
[root@argon pcardin]#  salt '*mac*' cmd.run "/opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/check-disk-usage.rb -w 97 -c 98 -p 'dev' &>/dev/null && echo 'worked' ; sysctl -n machdep.cpu.brand_string; uname -a"
mtl-mac-008:
    /bin/sh: line 1: 65943 Illegal instruction: 4  /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/check-disk-usage.rb -w 97 -c 98 -p 'dev' >&/dev/null
    Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
    Darwin mtl-mac-008 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
mtl-mac-012:
    worked
    Intel(R) Xeon(R) CPU E5-1620 v2 @ 3.70GHz
    Darwin mtl-mac-012 15.6.0 Darwin Kernel Version 15.6.0: Mon Aug 29 20:21:34 PDT 2016; root:xnu-3248.60.11~1/RELEASE_X86_64 x86_64
mtl-mac-004:
    worked
    Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
    Darwin mtl-mac-004 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
mtl-mac-003:
    worked
    Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
    Darwin mtl-mac-003 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
mtl-mac-011:
    worked
    Intel(R) Xeon(R) CPU E5-1620 v2 @ 3.70GHz
    Darwin mtl-mac-011 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
mtl-mac-007:
    /bin/sh: line 1: 64960 Illegal instruction: 4  /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/check-disk-usage.rb -w 97 -c 98 -p 'dev' >&/dev/null
    Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
    Darwin mtl-mac-007 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
mtl-mac-006:
    worked
    Intel(R) Xeon(R) CPU E5-1680 v2 @ 3.00GHz
    Darwin mtl-mac-006 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
mtl-mac-009:
    worked
    Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
    Darwin mtl-mac-009 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
mtl-mac-010:
    worked
    Intel(R) Xeon(R) CPU E5-1620 v2 @ 3.70GHz
    Darwin mtl-mac-010 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
mtl-mac-002:
    worked
    Intel(R) Xeon(R) CPU E5-1680 v2 @ 3.00GHz
    Darwin mtl-mac-002 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
oak-mac-003.atomicfiction.net:
    /bin/sh: line 1: 41010 Illegal instruction: 4  /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/check-disk-usage.rb -w 97 -c 98 -p 'dev' >&/dev/null
    Intel(R) Core(TM) i7 CPU         870  @ 2.93GHz
    Darwin oak-mac-003 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
oak-mac-002.atomicfiction.net:
    /bin/sh: line 1: 98809 Illegal instruction: 4  /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/check-disk-usage.rb -w 97 -c 98 -p 'dev' >&/dev/null
    Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
    Darwin oak-mac-002 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
oak-mac-006.atomicfiction.net:
    /bin/sh: line 1: 21507 Illegal instruction: 4  /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/check-disk-usage.rb -w 97 -c 98 -p 'dev' >&/dev/null
    Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
    Darwin oak-mac-006 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
oak-mac-008.atomicfiction.net:
    worked
    Intel(R) Xeon(R) CPU E5-1680 v2 @ 3.00GHz
    Darwin oak-mac-008 15.3.0 Darwin Kernel Version 15.3.0: Thu Dec 10 18:40:58 PST 2015; root:xnu-3248.30.4~1/RELEASE_X86_64 x86_64
mtl-mac-013:
    worked
    Intel(R) Xeon(R) CPU E5-1680 v2 @ 3.00GHz
    Darwin mtl-mac-013 15.6.0 Darwin Kernel Version 15.6.0: Thu Jun 23 18:25:34 PDT 2016; root:xnu-3248.60.10~1/RELEASE_X86_64 x86_64
```

Basically, looks like  Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, and Intel(R) Core(TM) i7 CPU         870  @ 2.93GHz don't work with the built-in ruby version provided by sensu... 

:(

**Comment by amdprophet** [comment]
Do the machines that Sensu runs on and the machines that Sensu doesn't run on have the same version of Mac OS? We've seen a similar issue with the old build tools for Mac - we should be able to set a flag or two to make the builds more accepting of CPU features.

**Comment by amdprophet** [comment]
Thanks! I was able to replicate the issue on some older hardware. Working on a fix.

**Comment by amdprophet** [comment]
This has been fixed in the latest builds (0.28.4). You can find download links on our downloads page https://sensuapp.org/downloads.
